src/averageAges.js

```diff
@@ -55,7 +61,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const child = people.filter(person => onlyWithSon
+    ? person.sex === 'm' && people.some(mother => mother.name === person.mother)
+    : people.some(mother => mother.name === person.mother));
+
+  const mothers = people.filter((person) => onlyWithSon
+    ? person.sex === 'f' && people.some(childs => person.name === childs.mother
+      && childs.sex === 'm')
+    : person.sex === 'f' && people.some(childs =>
+      person.name === childs.mother));
+
+  const ages = child.map(ch =>
+    ch.born - mothers.find(mother => mother.name === ch.mother).born);
```

Comment:
you can use people and delete get mothers (-1 cycle)

```suggestion
    ch.born - people.find(mother => mother.name === ch.mother).born);
```